portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,71 @@
 
 package com.google.sps.servlets;
 
+import java.util.List;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private static final String COMMENT_NAME = "name";
+  private static final String COMMENT_TEXT = "comment";
+  private static final String COMMENT_TIMESTAMP = "datePosted";
+
+  List<Comment> comments = new ArrayList<>();
+
+  @Override
+  public void init() {
+    // Retrieve stored comments
+    Query query = new Query("Comment").addSort(COMMENT_TIMESTAMP, SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    for(Entity entity : results.asIterable()) {
+      String text = (String) entity.getProperty(COMMENT_TEXT);
+      String name = (String) entity.getProperty(COMMENT_NAME);
+      long time = (long) entity.getProperty(COMMENT_TIMESTAMP);
+
+      comments.add(new Comment(text, name, time));
+    }
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Create and add the new comment
+    Comment comment = new Comment(request.getParameter(COMMENT_TEXT), 
```

Comment:
It doesn't feel necessary to create this Comment object since the only place it's used is when set entity properties.
Can use request.getParameter(COMMENT_TEXT) etc. to set the properties directly?
This way you don't even have to implement those comment.getXXX methods.

Reply:
Yes I agree - right now the `Comment` object isn't all that useful, and could be removed right now. Eventually though I plan on implementing some fancier applications for comments, which will take advantage of having a dedicated object and the methods I added (`compareTo`, `hashCode`).

Is that enough to keep the object for now or should I remove it for clarity?

Reply:
Yes, can keep if it's going to be used later.